repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala

```diff
@@ -396,4 +396,41 @@ class ReplSuite extends SparkFunSuite {
     Main.sparkContext.stop()
     System.clearProperty("spark.driver.port")
   }
+
+  test("register artifacts via SparkSession.addArtifact") {
```

Comment:
Can you use a UDF defined in the REPL?  If so how does this work with a JobArtifactSet? Do we layer the globally defined classpath over the session specific classpath? (I'd be nice to document this somewhere).

Reply:
I added one more test, which defines a UDF that initialises an external class added as an artifact.

> how does this work with a JobArtifactSet?

Can you elaborate? Afaik `JobArtifactSet` is not involved here since it's the artifact path that is applied when an active SparkSession is applied.

Classpath - It's the other way around: the session classpath is laid over the global one.